### PR TITLE
[PATCH API-NEXT v4] doc: userguide: ipsec state machine changes

### PR DESCRIPTION
**Pull request by Bill-Fischofer-Linaro**
Replace the FSMs used to describe SA state transitions and IPsec
operations with a Message Sequence Diagram (MSC) that shows the
same information in an easier to follow form.

Update User Guide to reflect these changes as well.

Signed-off-by: Bill Fischofer <bill.fischofer@linaro.org>

### CODE REVIEW

**Comment by Bill-Fischofer-Linaro** [comment]
This is an alternate IPsec doc update to follow the suggestion by @psavol to use a message sequence chart (MSC) rather than FSMs to explain the various SA and IPsec transitions and operating modes.

I'm not convinced this is clearer than the FSMs, but offered for consideration. Opinions welcome.

File: doc/users-guide/users-guide-ipsec.adoc [hunk 1]
```diff
@@ -244,12 +244,13 @@ IPsec operations may produce. This can be changed dynamically by the
...
-state diagram:
+In discussing the lifecycle of an SA and the operations it supports, it is
+useful to refer to the following sequence diagram for IPsec configuration, SA
+management, and IPsec operations:
 
-image::ipsec_fsm.svg[align="center"]
+image:ipsec_sa_states.svg[align="center"]
 
-After creation, IPsec services are active for this Security Association.  The
+After creation, IPsec services are active for this Ssecurity Association. The
```
Comment (lumag) [inline doc/users-guide/users-guide-ipsec.adoc]: typo

**Comment by Bill-Fischofer-Linaro** [comment]
Changes for v2:
* Rebased to latest api-next
* Corrected typo pointed out by @lumag

**Comment by Bill-Fischofer-Linaro** [comment]
Changes for v3:
* Corrected source file name from ipsec_sa_states.gv to ipsec_sa_states.msc to fix make distcheck

**Comment by muvarov** [comment]
#296 can be closed?

**Comment by Bill-Fischofer-Linaro** [comment]
@muvarov I'm not convinced this is an improvement over PR #296, which is why I did it as a separate PR. I'd like input from @psavol and Janne when they return from holiday.

**Comment by Bill-Fischofer-Linaro** [comment]
Ping to @psavol and @JannePeltonen

File: doc/images/ipsec_sa_states.msc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    p->a [label = "Done"];
+
+    a->o [label = "odp_ipsec_out()"];
+    o->p [label = "IPsec Encrypt"];
+    p->a [label = "Done"];
+
+    a->o [label = "odp_ipsec_out_inline()"];
+    o->p [label = "IPsec Encrypt Inline"];
+    p->o [label = "OK"];
+    o->a [label = "OK"];
```
Comment (JannePeltonen) [inline doc/images/ipsec_sa_states.msc]: Does the above perhaps give too strong an impression that odp_ipsec_out_inline() works synchronously (packet gets encrypted or even sent out before the _out_inline() returns)? Maybe the OK from ODP to app should precede the OK from platform to ODP?

**Comment by Bill-Fischofer-Linaro** [comment]
Changes for v3:
* Rebased to latest api-next
* Updated `odp_ipsec_out_inline()` sequence diagram as suggested by @JannePeltonen

**Comment by muvarov** [comment]
@JannePeltonen are you ok with this PR?

**Comment by JannePeltonen** [comment]
@muvarov  Yes, I am ok with this.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**